libs/langchain-community/src/vectorstores/pgvector.ts

```diff
@@ -506,22 +526,24 @@ export class PGVectorStore extends VectorStore {
    */
   async ensureCollectionTableInDatabase(): Promise<void> {
     try {
-      await this.pool.query(`
-        CREATE TABLE IF NOT EXISTS ${this.collectionTableName} (
+    const tableName = this.schemaName == null ? `"${this.tableName}"` : `"${this.schemaName}"."${this.tableName}"`
+      const queryString = `
+        CREATE TABLE IF NOT EXISTS "${this.collectionTableName}" (
           uuid uuid NOT NULL DEFAULT uuid_generate_v4() PRIMARY KEY,
           name character varying,
           cmetadata jsonb
         );
 
-        ALTER TABLE ${this.tableName}
+        ALTER TABLE ${tableName}
           ADD COLUMN collection_id uuid;
 
-        ALTER TABLE ${this.tableName}
-          ADD CONSTRAINT ${this.tableName}_collection_id_fkey
+        ALTER TABLE ${tableName}
+          ADD CONSTRAINT "${tableName.replaceAll('"', '')}_collection_id_fkey"
           FOREIGN KEY (collection_id)
-          REFERENCES ${this.collectionTableName}(uuid)
+          REFERENCES "${this.collectionTableName}(uuid)"
```

Comment:
This might be a syntax error?

Reply:
Good catch, those quotes should be removed now